src/tests/samples/violations/conditional_prefer_type_error.py

```diff
@@ -0,0 +1,176 @@
+"""
+Violation:
+
+Prefer TypeError when relevant.
+"""
+
+
+def incorrect_basic(some_arg):
+    if isinstance(some_arg, int):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_negation(some_arg):
+    if not isinstance(some_arg, int):
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_multiple_operators(some_arg):
+    if (False and (3 == 3)) or (not isinstance(some_arg, int) and True):
+        pass
+    else:
+        raise Exception("...")  # should be typeerror
+
+
+def incorrect_elif(some_arg):
```

Comment:
Great!

Reply:
This is not as obvious to me. I'd suggest to restricting where _all_ conditions consist of one of the type checking functions

Reply:
After reviewing it again, I agree with @sbrugman 

It's hard to tell the exception being raised there was caused by the type checking.

Reply:
Agreed!